core/src/web/camera.ts

```diff
@@ -18,36 +17,99 @@ export class CameraPluginWeb extends WebPlugin implements CameraPlugin {
   }
 
   async getPhoto(options: CameraOptions): Promise<CameraPhoto> {
-    options;
-
     return new Promise<CameraPhoto>(async (resolve, reject) => {
-      const cameraModal: any = document.createElement('pwa-camera-modal');
-      document.body.appendChild(cameraModal);
-      await cameraModal.componentOnReady();
-      cameraModal.addEventListener('onPhoto', async (e: any) => {
-        const photo = e.detail;
-
-        if (photo === null) {
-          reject('User cancelled photos app');
-        } else if (photo instanceof Error) {
-          reject(photo.message);
-        } else {
-          resolve(await this._getCameraPhoto(photo, options));
+      if (options.webUsePWAElements === true) {
+        const cameraModal: any = document.createElement('pwa-camera-modal');
+        document.body.appendChild(cameraModal);
+        try {
+          await cameraModal.componentOnReady();
+          cameraModal.addEventListener('onPhoto', async (e: any) => {
+            const photo = e.detail;
+
+            if (photo === null) {
+              reject('User cancelled photos app');
+            } else if (photo instanceof Error) {
+              reject(photo.message);
+            } else {
+              resolve(await this._getCameraPhoto(photo, options));
+            }
+
+            cameraModal.dismiss();
+            document.body.removeChild(cameraModal);
+          });
+
+          cameraModal.present();
+        } catch (e) {
+          this.fileInputExperience(options, resolve, reject);
         }
+      } else {
+        this.fileInputExperience(options, resolve, reject);
+      }
+    });
+  }
+
+  private fileInputExperience(options: CameraOptions, resolve: any, reject: any) {
+    let input = document.querySelector('#_capacitor-camera-input') as HTMLInputElement;
+
+    if (!input) {
+      input = document.createElement('input') as HTMLInputElement;
+      input.type = 'file';
+      input.accept = 'image/*';
```

Comment:
Wouldn't it make more sense to use `accept=".png, .jpg, .jpeg, .gif"` and remove the if stements below? In the comment `core/src/core-plugin-definitions.ts:356` you mentioned that only those are possible, so I think restrict the file input to exactly those makes more sense.

Reply:
Personally, when I specify extensions in the "accept" attribute the file selector doesn't even show up it would be a good idea to keep `image/*` until most browsers and WebView support the specific extensions.